site_scons/prereq_tools/base.py

```diff
@@ -1023,6 +1028,20 @@ def _resolve_patches(self):
                         '-o', patch_path, raw]]
             if not RUNNER.run_commands(command):
                 raise BuildFailure(raw)
+        # Remove old patches
+        for fname in os.listdir(self.patch_path):
+            if not fname.startswith(f"{self.name}_"):
+                continue
+            found = False
+            for key in patches.keys():
```

Comment:
> consider-iterating-dictionary, Consider iterating the dictionary directly instead of calling .keys()

```suggestion
            for key in patches:
```